app/core/Ledger/Ledger.ts

```diff
@@ -61,12 +60,12 @@ export const unlockLedgerDefaultAccount = async (
 }> => {
   const keyringController = Engine.context.KeyringController;
 
-  const keyring = await getLedgerKeyring();
-
-  if (isAccountImportReq) {
-    await keyringController.addNewAccountForKeyring(keyring);
-  }
-  const address = await keyring.getDefaultAccount();
+  const address = await withLedgerKeyring(async (keyring: LedgerKeyring) => {
+    if (isAccountImportReq) {
+      await keyringController.addNewAccountForKeyring(keyring);
```

Comment:
This will cause a deadlock, we can use `keyring.addAccounts(1)` instead since we don't use the idempotency feature of the `KeyringConctroller.addNewAccountForKeyring` method.

```suggestion
      await keyring.addAccounts(1);
```

(Probably it makes sense to remove `addNewAccountForKeyring` as a method from KeyringController in the future, since KeyringController doesn't even check if the passed keyring instance belongs to the controller)

Reply:
Ah great catch! Right, because `withKeyring` operations can be used to call write keyring methods, but not write KeyringController methods.

Reply:
I have committed this change, and added a warning to the JSDoc for this method to warn against this: https://github.com/MetaMask/metamask-mobile/pull/9570/commits/fede87fa5a554e3c207268b5e4be013b89306a89